pallets/dca/src/lib.rs

```diff
@@ -914,6 +923,10 @@ impl<T: Config> Pallet<T> {
 		schedule: &Schedule<T::AccountId, T::AssetId, BlockNumberFor<T>>,
 		amount_to_unreserve: Balance,
 	) -> DispatchResult {
+		if schedule.is_rolling() {
```

Comment:
strange that this is the only place this guard clause needed to be added, and not in `replan_or_complete` function, where the schedule is completed when there is not enough left in the RemainingAmounts.

But based on the unit tests, seems like working, so nice! An integration test might be handy here.